### PR TITLE
minor enhancements.

### PR DESCRIPTION
**Pull request by jortel**
Few minor API changes: better naming; cloning to better support mirror lists.

### CODE REVIEW

File: platform/pulpcore/download/single.py [hunk 1]
```diff
@@ -101,6 +102,25 @@ def register(self, event, handler):
...
+        """
+        Clone the download.
+        The clone has cleared:
+        - event handlers
+        - validations
+        - attachment
+        - reply
+
+        Returns:
+            Download: A cloned download.
```
Comment (mhrivnak) [inline platform/pulpcore/download/single.py]: Is this method intending to do the equivalent of:

```
clone = Download(self.url, self.writer)
clone.context = deepcopy(self.context)
return clone
```
?

I'm not suggesting that's better necessarily, but I'm trying to fully understand the intent. By doing a deepcopy, you'll end up with a copy of each attribute as well. You're dropping/resetting most of them, but leaving context as-is, which is an exact copy of the original.

File: platform/pulpcore/download/single.py [hunk 2]
```diff
@@ -101,6 +102,25 @@ def register(self, event, handler):
         """
         self.router.register(event, handler)
 
+    def clone(self):
```
Comment (mhrivnak) [inline platform/pulpcore/download/single.py]: Did you consider implementing this as ``__deepcopy__``?

See the last paragraph here for details: https://docs.python.org/3/library/copy.html#copy.deepcopy

File: platform/pulpcore/download/single.py (hunk 1 above)
Comment (jortel) [reply platform/pulpcore/download/single.py]: Not quite the same thing.

The clone() impl will return a copy of the *concrete* class with all of the attributes except those cleared or reset.  For example:
```
download = HttpDownload(..)
clone = download.clone()
```

The *clone* is an instance of HttpDownload with all of the certificates, proxy url etc copied. Actually, `clone()` needs to be adjusted to do a shallow copy of the *context*.

File: platform/pulpcore/download/single.py (hunk 2 above)
Comment (jortel) [reply platform/pulpcore/download/single.py]: Yes, but I think `clone()` has the following advantages:
* seems more explicit
* subclasses not required to impl anything.
* one less import for *user*.

Thoughts?

File: platform/pulpcore/download/single.py (hunk 1 above)
Comment (mhrivnak) [reply platform/pulpcore/download/single.py]: Gotcha, that makes sense. I figured you'd want the context to be shared, not copied, so thought I'd better get clarification.

The reminder that this is supposed to work with subclasses also makes a lot of sense.

File: platform/pulpcore/download/single.py (hunk 2 above)
Comment (mhrivnak) [reply platform/pulpcore/download/single.py]: +1 That makes sense, especially the part about making it work for subclasses.

**Comment by jortel** [comment]
@mhrivnak, updated so context shared not (deep) copied.  Anything else blocking?

**Pull request merged**